x-pack/elastic-agent/pkg/core/logger/logger.go

```diff
@@ -117,6 +118,7 @@ func makeInternalFileOutput(cfg *Config) (zapcore.Core, error) {
 		file.Interval(defaultCfg.Files.Interval),
 		file.RotateOnStartup(defaultCfg.Files.RotateOnStartup),
 		file.RedirectStderr(defaultCfg.Files.RedirectStderr),
+		file.Suffix(defaultCfg.Files.Suffix),
```

Comment:
defaultCfg here is from logp which is never Suffix, please pass `cfg.Files.Suffix` 
also when `filename` is specified in a way ^^ we end up with `elastic-agent-json.log-2021050416:38:16` 
which is weird. can we make it in a way where suffix is injected before `.log` file extension?

Reply:
This is the default behaviour of most log rotation tools. Also, this is not a new behaviour, we already append datetime to the end of the file when `interval` is configured. Furthermore, I had renamed `elastic-agent.log` to `elastic-agent` so we stay consistent in log names.